Makefile

```diff
@@ -185,121 +199,393 @@ test-integration: test-integration-aws test-integration-clickhouse test-integrat
+	sleep 10 # Many services are very slow... Give them a sec..
 endif
 	${MAYBE_ENVIRONMENT_EXEC} cargo test --no-default-features --features humio-integration-tests --lib ::humio:: -- --nocapture
 ifeq ($(AUTODESPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose stop
+	$(MAKE) -k stop-integration-humio
+endif
+
+start-integration-influxdb:
+ifeq ($(CONTAINER_TOOL),podman)
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) create --replace --name test-integration-influxdb -p 8086:8086 -p 9999:9999
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-influxdb --name influxdb_v1 \
+	 -e INFLUXDB_REPORTING_DISABLED=true influxdb:1.7
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-influxdb --name influxdb_v2 \
+	 -e INFLUXDB_REPORTING_DISABLED=true  quay.io/influxdb/influxdb:2.0.0-beta influxd --reporting-disabled
+else
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) create test-integration-influxdb
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-influxdb -p 8086:8086 --name influxdb_v1 \
+	 -e INFLUXDB_REPORTING_DISABLED=true influxdb:1.7
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-influxdb -p 9999:9999 --name influxdb_v2 \
+	 -e INFLUXDB_REPORTING_DISABLED=true  quay.io/influxdb/influxdb:2.0.0-beta influxd --reporting-disabled
 endif
 
+stop-integration-influxdb:
+	$(CONTAINER_TOOL) rm --force influxdb_v1 influxdb_v2 2>/dev/null; true
+ifeq ($(CONTAINER_TOOL),podman)
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) stop --name=test-integration-influxdb 2>/dev/null; true
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) rm --force --name test-integration-influxdb 2>/dev/null; true
+else
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) rm test-integration-influxdb 2>/dev/null; true
+endif
 
 test-integration-influxdb: ## Runs InfluxDB integration tests
 ifeq ($(AUTOSPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose up -d dependencies-influxdb
-	sleep 5 # Many services are very lazy... Give them a sec...
+	$(MAKE) -k stop-integration-influxdb \
+    ; rc=$$? \
+	$(MAKE) start-integration-influxdb
+	sleep 10 # Many services are very slow... Give them a sec..
 endif
 	${MAYBE_ENVIRONMENT_EXEC} cargo test --no-default-features --features influxdb-integration-tests --lib ::influxdb::integration_tests:: -- --nocapture
 ifeq ($(AUTODESPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose stop
+	$(MAKE) -k stop-integration-influxdb
+endif
+
+start-integration-kafka:
+ifeq ($(CONTAINER_TOOL),podman)
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) create --replace --name test-integration-kafka -p 2181:2181 -p 9091-9093:9091-9093
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-kafka --name zookeeper wurstmeister/zookeeper
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-kafka -e KAFKA_BROKER_ID=1 \
+	 -e KAFKA_ZOOKEEPER_CONNECT=zookeeper:2181 -e KAFKA_LISTENERS=PLAINTEXT://:9091,SSL://:9092,SASL_PLAINTEXT://:9093 \
```

Comment:
I wonder if it would be preferable to isolate these to an env file as per https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file ? We could put it in `scripts/integration-tests/`.